src/test/bridges/tokemak/TokemakBridge.t.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.10;
+
+import {BridgeTestBase} from "./../../aztec/base/BridgeTestBase.sol";
+
+// Example-specific imports
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IManager} from "../../../interfaces/tokemak/IManager.sol";
+
+import {TokemakBridge} from "../../../bridges/tokemak/TokemakBridge.sol";
+
+import {AztecTypes} from "../../../aztec/libraries/AztecTypes.sol";
+
+contract TokemakBridgeTest is BridgeTestBase {
+    // Vm constant vm = Vm(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
+    address public constant T_WETH = 0xD3D13a578a53685B4ac36A1Bab31912D2B2A2F36;
+    address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address public constant MANAGER = 0xA86e412109f77c45a3BC1c5870b880492Fb86A14;
+    address public constant DEPLOYER = 0x9e0bcE7ec474B481492610eB9dd5D69EB03718D5;
+
+    uint256 private nonce;
+    TokemakBridge private bridge;
+    AztecTypes.AztecAsset private empty;
+
+    uint256 private bridgeAddressId;
+
+    AztecTypes.AztecAsset private wAsset;
+    AztecTypes.AztecAsset private wtAsset;
+
+    event TokenBalance(uint256 previousBalance, uint256 newBalance);
+
+    function setUp() public {
+        bridge = new TokemakBridge(address(ROLLUP_PROCESSOR));
+        vm.deal(address(bridge), 0);
+        vm.startPrank(MULTI_SIG);
+
+        ROLLUP_PROCESSOR.setSupportedBridge(address(bridge), 2000000);
+
+        ROLLUP_PROCESSOR.setSupportedAsset(WETH, 100000);
+
+        ROLLUP_PROCESSOR.setSupportedAsset(T_WETH, 100000);
+
+        vm.stopPrank();
+
+        bridgeAddressId = ROLLUP_PROCESSOR.getSupportedBridgesLength();
+
+        wAsset = getRealAztecAsset(WETH);
+
+        wtAsset = getRealAztecAsset(T_WETH);
+    }
+
+    function testTokemakBridge() public {
+        validateTokemakBridge(1000, 500);
```

Comment:
The test should iterate over all the tokemak assets and the amounts should be fuzzed.